nodes/viz/texture_viewer.py

```diff
@@ -277,44 +362,80 @@ def save_bitmap(self, operator):
+        if img_format in format_mapping:
+            extension = '.' + format_mapping.get(img_format, img_format.lower())
+        else:
+            extension = '.' + img_format.lower()
         image_name = image_name + extension
         dim = size_tex_dict[self.selected_mode]
         width, height = dim, dim
-
         if image_name in bpy.data.images:
             img = bpy.data.images[image_name]
         else:
             img = bpy.data.images.new(name=image_name, width=width,
                                       height=height, alpha=alpha,
                                       float_buffer=True)
         img.scale(width, height)
-        np_buff = np.empty(len(img.pixels), dtype=np.float32)
-        np_buff.shape = (-1, 4)
-        np_buff[:, :] = np.array(buf)[:, np.newaxis]
-        np_buff[:, 3] = 1
-        np_buff.shape = -1
-        img.pixels[:] = np_buff
+        print('width is: {0}'.format(width))
+        print('length img pixels: {0}'.format(len(img.pixels)))
+        if col_mod == 'BW':
+            print("passing data from buf to pixels BW")
+            np_buff = np.empty(len(img.pixels), dtype=np.float32)
+            np_buff.shape = (-1, 4)
+            np_buff[:, :] = np.array(buf)[:, np.newaxis]
+            np_buff[:, 3] = 1
+            np_buff.shape = -1
+            img.pixels[:] = np_buff
+        elif col_mod == 'RGB':
+            print("passing data from buf to pixels RGB")
+            np_buff = np.empty(len(img.pixels), dtype=np.float32)
+            np_buff[:, :3] = np.array(buf)
+            # np_buff[:, 3] = 1
+            # np_buff.shape = -1
+            img.pixels[:] = np_buff
+        elif col_mod == 'RGBA':
+            print("passing data from buf to pixels RGBA")
+            # this works for RGBA!
+            img.pixels[:] = buf
 
         # get the scene context
         scene = bpy.context.scene
+        # set the scene quality to the maximum
+        scene.render.image_settings.quality = quality
+        # set different color depth
+        if img_format in {'JPEG', 'JPEG2000'}:
+            scene.render.image_settings.color_depth = '16'
+        else:
+            scene.render.image_settings.color_depth = '8'
+        # set compression level to no compression(0)
+        scene.render.image_settings.compression = compression
+        scene.render.image_settings.color_mode = col_mod
         scene.render.image_settings.file_format = img_format
+        print('settings done!')
         # get the path for the file and save the image
-        
-        desired_path = os.path.join(self.base_dir, self.image_name + extension)   #####
-
+        desired_path = os.path.join(self.base_dir, self.image_name + extension)
 
         img.save_render(desired_path, scene)
-        print('Saved!  path is:', desired_path)
+
+        print('Bitmap saved!  path is:', desired_path)
 
 
 def register():
+    bpy.utils.register_class(SvTextureVieverSliderCompression)
```

Comment:
i must delete i was tryng to add a slider with an operator, that is not  the rigth approach.